### PR TITLE
Respect `java_package` option in protobuf files

### PR DESCRIPTION
**Pull request by dmarticus**
Closes https://github.com/higherkindness/skeuomorph/issues/318.  Includes test cases provided by the issue filer :), plus some additional ones that cover missing package info altogether.

This PR is intended to give Skeuomorph comparable behavior to ScalaPB when it comes to handling the package structure -- I want to make it so that the package hierarchy goes like this (1) java package (if present) (2) package name (if present) (3) file name.  More on how ScalaPB does it here [here](https://scalapb.github.io/docs/generated-code/#default-package-structure).

### CODE REVIEW

File: src/main/scala/higherkindness/skeuomorph/mu/protocol.scala [hunk 1]
```diff
@@ -117,6 +117,9 @@ object Protocol {
       imports = protocol.imports.map(toImports)
     )
 
+//    println(s"the resultant protocol is $res")
+    res
+
```
Comment (juanpedromoreno) [inline src/main/scala/higherkindness/skeuomorph/mu/protocol.scala]: **suggestion**: Maybe we could revert the changes in this file :)

File: src/main/scala/higherkindness/skeuomorph/protobuf/schema.scala [hunk 2]
```diff
@@ -179,11 +179,12 @@ object ProtobufF {
       def makeOneOfB(oneOf: FieldF.OneOfField[A]): G[FieldF[B]] =
         f(oneOf.tpe).map(b => FieldF.OneOfField[B](oneOf.name, b, oneOf.indices): FieldF[B])
 
-      def traverseFieldF(fieldFList: List[FieldF[A]]): G[List[FieldF[B]]] =
+      def traverseFieldF(fieldFList: List[FieldF[A]]): G[List[FieldF[B]]] = {
         fieldFList.traverse {
           case field: FieldF.Field[A]      => makeFieldB(field).widen
           case oneOf: FieldF.OneOfField[A] => makeOneOfB(oneOf).widen
         }
+      }
```
Comment (juanpedromoreno) [inline src/main/scala/higherkindness/skeuomorph/protobuf/schema.scala]: **question**: why adding braces?

**Review by juanpedromoreno — APPROVED** [review]

File: src/main/scala/higherkindness/skeuomorph/mu/protocol.scala (hunk 1 above)
Comment (dmarticus) [reply src/main/scala/higherkindness/skeuomorph/mu/protocol.scala]: ooops I thought I did :)

File: src/main/scala/higherkindness/skeuomorph/protobuf/schema.scala (hunk 2 above)
Comment (dmarticus) [reply src/main/scala/higherkindness/skeuomorph/protobuf/schema.scala]: I _think_ this change is outdated?

File: src/main/scala/higherkindness/skeuomorph/protobuf/ParseProto.scala [hunk 3]
```diff
@@ -121,6 +121,34 @@ object ParseProto {
...
+  private def getPackageOrJavaPackage(file: FileDescriptorProto): String = {
+
+    val javaPackage: String = file.getOptions.getJavaPackage
+
+    val filePackage: String = file.getPackage
+
+    val fileName: String = formatName(file.getName)
+    // This package naming behavior is consistent with ScalaPB, see
+    // https://scalapb.github.io/docs/generated-code/#default-package-structure
+    if (javaPackage.isEmpty) {
```
Comment (dmarticus) [inline src/main/scala/higherkindness/skeuomorph/protobuf/ParseProto.scala:133]: Quick note here: I implemented this method to stay consistent with the parsing behavior of ScalaPB.  If we want to make it so that we don’t use the filename on the absence of any package information (which would fail the parsing step), I’m open to that as well.  At this point, without better direction, I'm just going for compatibility with an existing solution.

File: src/main/scala/higherkindness/skeuomorph/protobuf/ParseProto.scala [hunk 4]
```diff
@@ -121,6 +121,34 @@ object ParseProto {
...
+  private def getTypeNameFromQualifiedName(fullName: String): String =
+    // When comparing the types, we only need to know the name of the type, not the
+    // fully-qualified name (since we use the value of the package passed into the protoc to create
+    // the TNamedType.  However, depending on how we access the type name via `getTypeName`, there is
+    // different behavior depending on whether we access a type name from a file with a `package` header
+    // (which will return the fully-qualified name) vs accessing a type name from a file with a `java_package`
+    // header (which will just return the name of the type).  This method trims the type name from a fully-
+    // qualified name if necessary, which provides consistent behavior when comparing the name of a message
+    // to the name of a type
+    fullName.substring(fullName.lastIndexOf(".") + 1)
```
Comment (dmarticus) [inline src/main/scala/higherkindness/skeuomorph/protobuf/ParseProto.scala:150]: Another long comment that basically explains the caveat here: we no longer will be comparing fully-qualified type names to the type that appears in the method, because [getTypeName](https://www.codota.com/code/java/methods/com.google.protobuf.DescriptorProtos$FieldDescriptorProto/getTypeName) behaves differently depending on what it's called against (if called against a file with a `package` header, it returns the fully-qualified type, whereas if called against a file with a `java_package` header, it just returns the typeName).  In this case, we always want to just use the typename, so we parse the string down if necessary.  Yes, I know it's ugly "strings as types" but when working with the java methods that only return strings there's not a lot I can do about it :|

File: src/test/scala/higherkindness/skeuomorph/protobuf/rating.proto [hunk 5]
```diff
@@ -1,6 +1,6 @@
 syntax = "proto3";
 
-package com.acme;
+option java_package = "com.acme";
```
Comment (dmarticus) [inline src/test/scala/higherkindness/skeuomorph/protobuf/rating.proto:3]: testing that the `java_package` interop works seamlessly when importing proto files (spoiler: it does)

File: src/main/scala/higherkindness/skeuomorph/avro/Protocol.scala [hunk 6]
```diff
@@ -25,7 +25,7 @@ import org.apache.avro.{Schema, Protocol => AvroProtocol}
 import higherkindness.droste._
 import higherkindness.droste.syntax.all._
 
-import scala.jdk.CollectionConverters._
+import scala.collection.JavaConverters._
```
Comment (dmarticus) [inline src/main/scala/higherkindness/skeuomorph/avro/Protocol.scala:28]: fwiw the explanation on why I'm making this change is nicely laid out by Birchall here: https://github.com/higherkindness/skeuomorph/pull/322#issuecomment-682474294

**Review by juanpedromoreno — APPROVED** [review]
Great job @dmarticus!
